proto/src/constants.rs

```diff
@@ -6,3 +6,144 @@ pub const APPLICATION_JSON: &str = "application/json";
+pub const ATTR_NAME: &str = "name";
+pub const ATTR_NO_INDEX: &str = "no-index";
+pub const ATTR_NSUNIQUEID: &str = "nsuniqueid";
+pub const ATTR_OAUTH2_ALLOW_INSECURE_CLIENT_DISABLE_PKCE: &str =
+    "oauth2_allow_insecure_client_disable_pkce";
+pub const ATTR_OAUTH2_CONSENT_SCOPE_MAP: &str = "oauth2_consent_scope_map";
+pub const ATTR_OAUTH2_JWT_LEGACY_CRYPTO_ENABLE: &str = "oauth2_jwt_legacy_crypto_enable";
+pub const ATTR_OAUTH2_PREFER_SHORT_USERNAME: &str = "oauth2_prefer_short_username";
+pub const ATTR_OAUTH2_RS_BASIC_SECRET: &str = "oauth2_rs_basic_secret";
+pub const ATTR_OAUTH2_RS_IMPLICIT_SCOPES: &str = "oauth2_rs_implicit_scopes";
+pub const ATTR_OAUTH2_RS_NAME: &str = "oauth2_rs_name";
+pub const ATTR_OAUTH2_RS_ORIGIN_LANDING: &str = "oauth2_rs_origin_landing";
+pub const ATTR_OAUTH2_RS_ORIGIN: &str = "oauth2_rs_origin";
+pub const ATTR_OAUTH2_RS_SCOPE_MAP: &str = "oauth2_rs_scope_map";
+pub const ATTR_OAUTH2_RS_SUP_SCOPE_MAP: &str = "oauth2_rs_sup_scope_map";
+pub const ATTR_OAUTH2_RS_TOKEN_KEY: &str = "oauth2_rs_token_key";
+pub const ATTR_OAUTH2_SESSION: &str = "oauth2_session";
+pub const ATTR_OBJECTCLASS: &str = "objectclass";
+pub const ATTR_OTHER_NO_INDEX: &str = "other-no-index";
+pub const ATTR_PASSKEYS: &str = "passkeys";
+pub const ATTR_PASSWORD_IMPORT: &str = "password_import";
+pub const ATTR_PHANTOM: &str = "phantom";
+pub const ATTR_PRIMARY_CREDENTIAL: &str = "primary_credential";
+pub const ATTR_PRIVATE_COOKIE_KEY: &str = "private_cookie_key";
+pub const ATTR_RADIUS_SECRET: &str = "radius_secret";
+pub const ATTR_RECYCLED: &str = "recycled";
+pub const ATTR_REPLICATED: &str = "replicated";
+pub const ATTR_RS256_PRIVATE_KEY_DER: &str = "rs256_private_key_der";
+pub const ATTR_SELF: &str = "self";
+pub const ATTR_SOURCE_UUID: &str = "source_uuid";
+pub const ATTR_SPN: &str = "spn";
+pub const ATTR_SSH_PUBLICKEY: &str = "ssh_publickey";
+pub const ATTR_SSHPUBLICKEY: &str = "sshpublickey";
+pub const ATTR_SYNC_ALLOWED: &str = "sync_allowed";
+pub const ATTR_SYNC_COOKIE: &str = "sync_cookie";
+pub const ATTR_SYNC_CREDENTIAL_PORTAL: &str = "sync_credential_portal";
+pub const ATTR_SYNC_PARENT_UUID: &str = "sync_parent_uuid";
+pub const ATTR_SYNC_TOKEN_SESSION: &str = "sync_token_session";
+pub const ATTR_SYNC_YIELD_AUTHORITY: &str = "sync_yield_authority";
+pub const ATTR_SYNTAX: &str = "syntax";
+pub const ATTR_SYSTEMMAY: &str = "systemmay";
+pub const ATTR_SYSTEMMUST: &str = "systemmust";
+pub const ATTR_TERM: &str = "term";
+pub const ATTR_UID: &str = "uid";
+pub const ATTR_UIDNUMBER: &str = "uidnumber";
+pub const ATTR_UNIQUE: &str = "unique";
+pub const ATTR_UNIX_PASSWORD: &str = "unix_password";
+pub const ATTR_USER_AUTH_TOKEN_SESSION: &str = "user_auth_token_session";
+pub const ATTR_USERID: &str = "userid";
+pub const ATTR_USERPASSWORD: &str = "userpassword";
+pub const ATTR_UUID: &str = "uuid";
+pub const ATTR_VERSION: &str = "version";
+
+pub const TEST_ATTR_NON_EXIST: &str = "non-exist";
```

Comment:
Could these be cfg!(test) if they are test only?

Reply:
There's something weird with constants - if I set any of the various `#[cfg(test)]` / `#[cfg(any(test, debug_assertions))]` etc it won't find it when running `cargo test`

Reply:
Hmmm okay. Maybe we TODO flag this then.